### PR TITLE
Add hooks for gamm and lockup modules

### PR DESCRIPTION
**Pull request by sunnya97**
Cherrypicked addition of hooks to x/gamm and x/lockup from @Thunnini's work in #80

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -319,8 +319,20 @@ func NewOsmosisApp(
...
 	app.EvidenceKeeper = *evidenceKeeper
 
-	app.GAMMKeeper = gammkeeper.NewKeeper(appCodec, keys[gammtypes.StoreKey], app.AccountKeeper, app.BankKeeper)
-	app.LockupKeeper = *lockupkeeper.NewKeeper(appCodec, keys[lockuptypes.StoreKey], app.AccountKeeper, app.BankKeeper)
+	gammKeeper := gammkeeper.NewKeeper(appCodec, keys[gammtypes.StoreKey], app.AccountKeeper, app.BankKeeper)
+	lockupKeeper := lockupkeeper.NewKeeper(appCodec, keys[lockuptypes.StoreKey], app.AccountKeeper, app.BankKeeper)
+
+	app.GAMMKeeper = *gammKeeper.SetHooks(
+		gammtypes.NewMultiGammHooks(
+		// insert gamm hooks receivers here
```
Comment (ValarDragon) [inline app/app.go:327]:
```suggestion
		// TODO: insert gamm hooks receivers here
```

File: app/app.go [hunk 2]
```diff
@@ -319,8 +319,20 @@ func NewOsmosisApp(
...
+		),
+	)
+
+	app.LockupKeeper = *lockupKeeper.SetHooks(
+		lockuptypes.NewMultiLockupHooks(
+		// insert lockup hooks receivers here
```
Comment (ValarDragon) [inline app/app.go:333]:
```suggestion
		// TODO: insert lockup hooks receivers here
```

**Review by ValarDragon — APPROVED** [review]
LGTM

File: app/app.go (hunk 1 above)
Comment (sunnya97) [reply app/app.go:327]: It's not really a TODO, as much of a "if something needs these hooks, this is where they go".  Nothing uses them yet.

**Comment by ValarDragon** [comment]
Update: We're redoing the changes to actually git cherry-pick the history, to get proper accreditation for commit authorship

**Comment by sunnya97** [comment]
Fixed 👍

**Review by ValarDragon — APPROVED** [review]
LGTM

**Comment by Thunnini** [comment]
The reason why the hook was added to the lockup module was because the pool-yield module needed a hook, and it wasn’t implemented at the time. It was mostly a temporary measure. @antstalepresh If you’ve implemented it now, then I think Eugen’s code should be merged. What do you think?

**Review by antstalepresh — APPROVED** [review]

**Comment by antstalepresh** [comment]
> The reason why the hook was added to the lockup module was because the pool-yield module needed a hook, and it wasn’t implemented at the time. It was mostly a temporary measure. @antstalepresh If you’ve implemented it now, then I think Eugen’s code should be merged. What do you think?

I didn't have time to write hooks for lockup module, was focusing on incentives module. But it seems good, already have 2 hooks added to the codebase.

**Pull request merged**